design/one-pager-configuration-stacks.md

```diff
@@ -254,30 +244,35 @@ sense for developer to have a folder for each tier under `resources` folder.
 ### Reconciliation
 
 There are mainly 3 responsibility of the controller:
-* When all deployed resources are ready, `status.state` should be `ready`.
-  * For resource classes; if the resource exists in the api-server it's considered
-    to be ready.
-  * For managed resources; if the `Ready` condition status is `true`, then it's
-    considered to be ready.
-  * For other possible resources, the developer chooses what to look for as
-    readiness.
-* Apply; continuously reproduce the resources from YAMLs and apply it.
-* Deletion: should be done via label selector when the CR is deleted.
-
-In the `Apply` phase, there will be some processing phasse before calling `Create`
+* Creation: Generate the resources from YAMLs and create them.
+* Apply: Continuously reproduce the resources from YAMLs and update.
+* Deletion: When the CR is deleted, all deployed resources should be deleted
+  via owner reference relations.
+
+In the `Apply` phase, there will be some processing phase before calling `Create`
 for the resources:
 
-1. Names of the resources should be converted to the following format
-  `<CR namespace>-<CR name>-<Name given in YAML>` to prevent name collisions
-  since most of the resources stack deploys are cluster-scoped.
-  So, the name that appears on YAML files are valid for referencing each other
+1. Names of the resources should be converted to the processing phase following
+  format `<CR name>-<Name given in YAML>` to prevent name collisions.
```

Comment:
Is there something we can do to make it more obvious that this transformation will happen? I worry that it could be surprising.

It seems like the stack author will write something like:

```yaml
providerRef:
  name: cool-provider
```

But the resource we actually create will be modified to:

```yaml
providerRef:
  name: mycoolcustomresource-cool-provider
```